sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -56,16 +58,18 @@ def test_silhouette():
             assert_almost_equal(score_euclidean, score_dense_with_sampling)
 
 
-def test_no_nan():
+def test_cluster_size_1():
     # Assert Silhouette Coefficient != nan when there is 1 sample in a class.
-    # This tests for the condition that caused issue 960.
-    # Note that there is only one sample in cluster 0. This used to cause the
-    # silhouette_score to return nan (see bug #960).
+    # The original silhouette paper suggests that while arbitrary,
+    # single-sample clusters should receive a score of 0.
     labels = np.array([1, 0, 1, 1, 1])
     # The distance matrix doesn't actually matter.
     D = np.random.RandomState(0).rand(len(labels), len(labels))
     silhouette = silhouette_score(D, labels, metric='precomputed')
     assert_false(np.isnan(silhouette))
+    ss = silhouette_samples(D, labels, metric='precomputed')
+    assert_false(np.isnan(ss).any())
+    assert_equal(ss[1], 0.)
```

Comment:
Instead of using a random precomputed metric array, could you please use a toy 2D X array with shape (4, 1) in 1D euclidean spaces such as `X = [[1., 0., 1., 2.]]` and then check that the output of `silhouette_samples` is `[1., 0., 1., 0.]` as expected?